src/components/shared/Promotions/Promotions.js

```diff
@@ -0,0 +1,47 @@
+import React, { useState, useEffect } from 'react';
+import { injectIntl } from 'react-intl';
+import { InjectAppServices } from '../../../services/pure-di';
+import Loading from '../../Loading/Loading';
+
+/**
+ * Promotions
+ * @param { Object } props
+ * @param { import('react-intl').InjectedIntl } props.intl
+ * @param { import('../../services/pure-di').AppServices } props.dependencies
+ */
+const Promotions = function({ intl, type, page, dependencies: { dopplerLegacyClient } }) {
+  const [bannerData, setBannerData] = useState({});
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      setIsLoading(true);
+      const bannerData = await dopplerLegacyClient.getBannerData(intl.locale, type, page || '');
+      setBannerData(bannerData);
+      setIsLoading(false);
+    };
+
+    fetchData();
+  }, []);
```

Comment:
What do you think about refactoring it as a hook?

```js
const [bannerData, isLoading] = useFetchBannerData({ locale: int.locale, type, page });
```

It is not so much valuable because it is used only once, but it is tydy and maybe it is also fun.

Reply:
i dont understand how is work this

Reply:
Estoy proponiendo crear un Hook con esa interface. No se gana tanto porque no se va a reusar, la ventaja es que simplifica el código del componente.

Si te interesa pero sigo sin explicarme bien, llamame y lo hacemos juntos.

Si no te interesa, me parece bien, no es algo tan necesario.

Reply:
dale primero voy a ver bien lo demas del otro servicio pero me interesa tb ver como se hace, parece interesante